### PR TITLE
Minor fixes for Monitoring v2

### PR DESCRIPTION
**Pull request by aiyengar2**
Related Issues: https://github.com/rancher/rancher/issues/30712, https://github.com/rancher/rancher/issues/30191, https://github.com/rancher/rancher/issues/30684, https://github.com/rancher/rancher/issues/30680

### CODE REVIEW

File: packages/rancher-monitoring/generated-changes/patch/charts/prometheus-node-exporter/values.yaml.patch [hunk 1]
```diff
@@ -9,3 +9,12 @@
    tag: v1.0.1
    pullPolicy: IfNotPresent
  
+@@ -98,6 +98,8 @@
+ tolerations:
+   - effect: NoSchedule
+     operator: Exists
++  - effect: NoExecute
```
Comment (aiyengar2) [inline packages/rancher-monitoring/generated-changes/patch/charts/prometheus-node-exporter/values.yaml.patch:16]: https://github.com/rancher/rancher/issues/30680#issuecomment-754850258

File: packages/rancher-monitoring/generated-changes/patch/Chart.yaml.patch [hunk 2]
```diff
@@ -11,6 +11,8 @@
 +  catalog.cattle.io/provides-gvr: monitoring.coreos.com.prometheus/v1
 +  catalog.cattle.io/display-name: "Monitoring"
 +  catalog.cattle.io/auto-install: rancher-monitoring-crd=match
++  catalog.cattle.io/requests-cpu: "4500m"
++  catalog.cattle.io/requests-memory: "4000Mi"
```
Comment (aiyengar2) [inline packages/rancher-monitoring/generated-changes/patch/Chart.yaml.patch:15]: Breakdown for why these values were chosen is found in https://github.com/rancher/rancher/issues/30684#issuecomment-765648833

File: packages/rancher-pushprox/charts/templates/_helpers.tpl [hunk 3]
```diff
@@ -27,7 +27,7 @@ provider: kubernetes
 {{- if .Values.clients.proxyUrl -}}
 {{ printf "%s" .Values.clients.proxyUrl }}
 {{- else -}}
-{{ printf "http://%s.%s.svc.cluster.local:%d" (include "pushProxy.proxy.name" .) .Release.Namespace (int .Values.proxy.port) }}
+{{ printf "http://%s.%s.svc:%d" (include "pushProxy.proxy.name" .) .Release.Namespace (int .Values.proxy.port) }}
```
Comment (aiyengar2) [inline packages/rancher-pushprox/charts/templates/_helpers.tpl:30]: .cluster.local is unnecessary

https://github.com/rancher/rancher/issues/30191#issuecomment-744758937

File: packages/rancher-monitoring/generated-changes/patch/templates/prometheus/prometheus.yaml.patch [hunk 4]
```diff
@@ -28,3 +28,16 @@
  {{- end }}
  {{- if .Values.prometheus.prometheusSpec.initContainers }}
    initContainers:
+@@ -242,6 +245,7 @@
+ {{- if .Values.prometheus.prometheusSpec.disableCompaction }}
+   disableCompaction: {{ .Values.prometheus.prometheusSpec.disableCompaction }}
+ {{- end }}
++{{- if .Values.prometheus.prometheusSpec.portName }}
```
Comment (aiyengar2) [inline packages/rancher-monitoring/generated-changes/patch/templates/prometheus/prometheus.yaml.patch:35]: https://github.com/rancher/rancher/issues/30712

**Review by prachidamle — APPROVED** [review]
LGTM

**Review by dramich — APPROVED** [review]

**Pull request merged**